src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Use maxLengthEnforcement instead of maxLengthEnforced
+description: Introducing the MaxLengthEnforcement enum.
+---
+
+## Summary
+
+_If you don't use the `maxLength` or `maxLengthEnforced` properties in a
+`TextField`, `TextFormField`, or `CupertinoTextField`,
+or you don't care about the truncation behavior of an
+editable text field, you can stop reading._
+
+To control the behavior of `maxLength`
+in the `LengthLimitingTextInputFormatter`,
+use `maxLengthEnforcement` instead of `maxLengthEnforced`, which is deprecated.
+
+## Context
+
+Before [PR 63754][]: Fix TextField crashed with composing and maxLength set,
+the `LengthLimitingTextInputFormatter` always truncated the value in the text
+field, which resulted in engine crashes when the composing range sometimes
+overflowed. With this change, an editable `TextEditingValue` is not truncated
+if only `maxLength` is set, but _is_ truncated after editing is complete.
+
+With [PR 68086][]: Introduce `MaxLengthEnforcement`,
+the truncation behavior is now controlled by passing
+`maxLengthEnforcement` to the `LengthLimitingTextInputFormatter`,
+which is then exposed to the `TextField`, `TextFormField`,
+and `CupertinoTextField` widgets.
+The `maxLengthEnforced` property is now deprecated.
+(For more information, see the docs for [`maxLength`][] and
+[`MaxLengthEnforcement`][].)
+
+The default max length truncation behavior is based on the `TargetPlatform`
+that you're using or running on.
+
+## Description of change
+
+* Added a `MaxLengthEnforcement` enum, which controls all max length
+  limiting behavior.
+* The `maxLengthEnforced` property in editable text field classes is deprecated.
+* A named optional parameter, `maxLengthEnforcement` is added to
+  the `TextField`, `TextFormField`, `CupertinoTextField`,
+  and `LengthLimitingTextInputFormatter` classes.
+* The default behavior for different platforms is updated, including Android,
+  iOS, Web, Windows, MacOS, Linux, and Fuchsia.
+
+## Migration guide
+
+The following tips describe how to produce common behaviors, and which behavior
```

Comment:
Maybe replace this paragraph with one that describes how every value in the deprecated parameter map to the new parameter (maybe with a table?), and when should the new `truncateAfterCompositionEnds` value be used, for people that just want to move away from the deprecated parameter.

Reply:
There might be too many variable for "when should use which parameter", such as "CJK or not", "Gboard or not", "Which platform is"...

Reply:
But I think we should recommend users to use platforms default behavior in order to avoid breaking someone that is not propose to input like the behavior that the developer has used.

Reply:
Got it

Reply:
> Makes sense to me. But I feel we shouldn't "recommend" the default behavior since we want developers to pick the right behavior for their use cases.

We make different platforms have their default behavior because we want the behavior is the same as the original native platform, aren't we? So why we shouldn't recommend it? IMO the goal is developers don't need to choose which behavior they should use, but satisfied with the default behavior.

Reply:
Although this is the migration guide, which means people who's using the `maxLengthEnforced` might read this guide, there will be more people that may found different behaviors and come here to see the guide.